src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -69,10 +69,10 @@ public class BodyModelBinder : BindingSourceModelBinder
             {
                 model = GetDefaultValueForType(bindingContext.ModelType);
                 bindingContext.ModelState.AddModelError(bindingContext.ModelName, ex);
-                return new ModelBindingResult(model: null, key: bindingContext.ModelName, isModelSet: false);
+                return new ModelBindingResult(model: null, key: string.Empty, isModelSet: false);
```

Comment:
Lets return bindingContext.ModelName when we fail, In this case since we ourselves are adding model state error we should return that key.